### PR TITLE
[move-prover] Add remaining loop invariants to vector functions

### PR DESCRIPTION
**Pull request by shazqadeer**
## Motivation

Finish verifying the loops in vector.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

No code change.  Move Prover is verifying the Move file.

### CODE REVIEW

File: language/move-prover/tests/sources/functional/verify_vector.move [hunk 1]
```diff
@@ -145,16 +146,33 @@ module VerifyVector {
...
+            Vector::push_back(v, Vector::pop_back(&mut other))
+        };
         Vector::destroy_empty(other);
     }
-    spec fun verify_append { // TODO: cannot verify loop
-        //ensures len(lhs) == old(len(lhs) + len(other)); //! A postcondition might not hold on this return path.
-        //ensures lhs[0..len(old(lhs))] == old(lhs); //! A postcondition might not hold on this return path.
-        //ensures lhs[len(old(lhs))..len(lhs)] == old(other); //! A postcondition might not hold on this return path.
+    spec fun verify_append {
+        ensures len(v) == old(len(v) + len(other)); //! A postcondition might not hold on this return path.
```
Comment (junkil-park) [inline language/move-prover/tests/sources/functional/verify_vector.move:173]: Could you remove the comment which was the previous error message?

File: language/move-prover/tests/sources/functional/verify_vector.move [hunk 2]
```diff
@@ -280,12 +298,10 @@ module VerifyVector {
         Vector::pop_back(v)
     }
     spec fun verify_remove {
-        // TODO: cannot verify all loop invariants
-        // TODO: input parameter must not be modified to enable referring to its value at entry
         aborts_if j >= len(v); //! A postcondition might not hold on this return path.
         ensures len(v) == len(old(v)) - 1; //! A postcondition might not hold on this return path.
         ensures v[0..j] == old(v[0..j]); //! A postcondition might not hold on this return path.
-        //ensures v[j..len(v)] == old(v[j+1..len(v)]); //! A postcondition might not hold on this return path.
+        ensures v[j..len(v)] == old(v[j+1..len(v)]); //! A postcondition might not hold on this return path.
```
Comment (junkil-park) [inline language/move-prover/tests/sources/functional/verify_vector.move:304]: The comment in this line as well.

Comment (shazqadeer) [reply language/move-prover/tests/sources/functional/verify_vector.move]: Done.

File: language/move-prover/tests/sources/functional/verify_vector.move (hunk 1 above)
Comment (shazqadeer) [reply language/move-prover/tests/sources/functional/verify_vector.move]: Done.

**Pull request closed without merging**